crates/services/executor/src/executor.rs

```diff
@@ -738,66 +654,126 @@ where
             .read_transaction()
             .with_policy(ConflictPolicy::Overwrite);
 
-        debug_assert!(block.transactions.is_empty());
+        debug_assert!(partial_block.transactions.is_empty());
+
+        self.process_relayed_txs(
+            forced_transactions,
+            &mut partial_block,
+            &mut thread_block_transaction,
+            &mut data,
+            coinbase_contract_id,
+        )?;
+
+        for transaction in transactions {
+            let maybe_checked_tx =
+                MaybeCheckedTransaction::Transaction(transaction.clone());
+            let _new_transaction = self.execute_transaction_and_commit(
+                &mut partial_block,
+                &mut thread_block_transaction,
+                &mut data,
+                maybe_checked_tx,
+                gas_price,
+                coinbase_contract_id,
+            )?;
+            #[cfg(debug_assertions)]
+            if _new_transaction != transaction {
+                error!("Provided transaction does not match the transaction built by the executor");
+            }
```

Comment:
I'm not sure I really like this. It's kinda chatty in the tests. Is there another approach we could take? We can't have it throw an error because the tests are expecting the error that will be used in prod.

Reply:
Instead of comparing transactions here, we can do that inside of the `check_block_matches`. You can even iterate over the block's transactions there and compare transactions and have a behavior like before the change(return `InvalidTransactionOutcome`)=)